golos.publication/golos.publication.cpp

```diff
@@ -613,8 +614,17 @@ void publication::set_vote(name voter, name author, string permlink, int16_t wei
     std::vector<structures::delegate_voter> delegators;
     auto token_code = pool->state.funds.symbol.code();
     auto list_delegate_voter = golos::vesting::get_list_delegate(config::vesting_name, voter, token_code);
-    for (auto record : list_delegate_voter) 
-        delegators.push_back( {record.sender, record.quantity} );
+    auto effective_vesting = golos::vesting::get_account_effective_vesting(config::vesting_name, voter, token_code);
+    
+    for (auto record : list_delegate_voter) {
+        auto interest_rate = static_cast<uint16_t>(static_cast<uint128_t>(record.quantity.amount) * 
+                    record.interest_rate / effective_vesting.amount);
```

Comment:
Before converting to uint16_t, we need to add a assert that the resulting value is no more than 10,000.

@zxcat 0.01% accuracy may be insufficient and a delegator with a small amount of delegation will never receive an award. This is Ok?

Reply:
@s-medvedev, yes, it's ok, this removes low-value bots and reduces required computations

Reply:
@s-medvedev, here is impossible to get >100%: `record.quantity` is always less than `effective_vesting` because it's part of effective vesting